pkg/devfile/parser/context/context.go

```diff
@@ -90,11 +104,24 @@ func (d *DevfileCtx) Populate() (err error) {
 
 // PopulateFromURL fills the DevfileCtx struct with relevant context info
 func (d *DevfileCtx) PopulateFromURL() (err error) {
-
-	_, err = url.ParseRequestURI(d.url)
+	u, err := url.ParseRequestURI(d.url)
 	if err != nil {
 		return err
 	}
+	if !strings.HasSuffix(d.url, ".yaml") {
+		u.Path = path.Join(u.Path, "devfile.yaml")
+		param := util.HTTPRequestParams{
+			URL: u.String(),
+		}
+		if _, err = util.HTTPGetRequest(param, 0); err != nil {
```

Comment:
can you confirm if there is an issue? I am just running the main executable. Just added one line in my local branch `fmt.Printf("parsing devfile with %+v\n", args)`. You can visit that raw link and I can access it from my browser 🤔 

It seems to be fine with local path.

```
Maysuns-MacBook-Pro:parser maysun$ ./main https://raw.githubusercontent.com/maysunfaisal/parser/testdot
parsing devfile from https://raw.githubusercontent.com/maysunfaisal/parser/testdot
parsing devfile with {Path: URL:https://raw.githubusercontent.com/maysunfaisal/parser/testdot Data:[] FlattenedDevfile:<nil> RegistryURLs:[]}
the provided url is not a valid yaml filepath, and no devfile.yaml or .devfile.yaml under provided url: https://raw.githubusercontent.com/maysunfaisal/parser/testdot
Maysuns-MacBook-Pro:parser maysun$ 



Maysuns-MacBook-Pro:parser maysun$ 
Maysuns-MacBook-Pro:parser maysun$ ./main https://raw.githubusercontent.com/maysunfaisal/parser/testdot/.devfile.yaml
parsing devfile from https://raw.githubusercontent.com/maysunfaisal/parser/testdot/.devfile.yaml
parsing devfile with {Path: URL:https://raw.githubusercontent.com/maysunfaisal/parser/testdot/.devfile.yaml Data:[] FlattenedDevfile:<nil> RegistryURLs:[]}
schema version: 2.1.0
All component: 
runtime
hehehehe
runtime3
runtime4
All Exec commands: 
command install is with kind: buildworkingDir is: /project-starter
command run is with kind: runworkingDir is: /project
command debug is with kind: debugworkingDir is: /project
command test is with kind: testworkingDir is: /project
command install2 is with kind: buildworkingDir is: /project
command run2 is with kind: runworkingDir is: /project
command debug2 is with kind: debugworkingDir is: /project
command test2 is with kind: testworkingDir is: /project
=========================================================
Container components applied filter: 
hehehehe
Exec commands applied filter: 
command install2 is with kind: buildworkingDir is: /project
command run2 is with kind: runworkingDir is: /project
dockerfilePath: /relative/path/to/Dockerfile
```

Reply:
Adding to this, could you also updated an unrelated err msg - it says parent which is wrong and confusing ->

```
Maysuns-MacBook-Pro:parser maysun$ ./main https://raw.githubusercontent.com/maysunfaisal/parser/testdot2/devfile.yaml
args [./main https://raw.githubusercontent.com/maysunfaisal/parser/testdot2/devfile.yaml]
parsing devfile from https://raw.githubusercontent.com/maysunfaisal/parser/testdot2/devfile.yaml
error getting parent info from url: fail to retrive https://raw.githubusercontent.com/maysunfaisal/parser/testdot2/devfile.yaml: Not Found
```

Reply:
It is very wired that sending a GET request to the url you tested, receives `404: not found`
<img width="708" alt="Screen Shot 2021-03-18 at 5 38 36 PM" src="https://user-images.githubusercontent.com/22285490/111701250-ec491980-8810-11eb-8c63-bd36ea2a5055.png">

Reply:
updated